homeassistant/components/fritz/binary_sensor.py

```diff
@@ -17,79 +24,91 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+def _retrieve_connection_state(status: FritzStatus) -> bool:
+    """Return download line attenuation."""
+    return bool(status.is_connected)
+
+
+def _retrieve_link_state(status: FritzStatus) -> bool:
+    """Return download line attenuation."""
+    return bool(status.is_linked)
+
```

Comment:
There is no reason to have these out of the class.

Just do this in the update:
```python
if self.entity_descriotion.key == "is_connected":
	self._attr_is_on = bool(status.is_connected)
elif self.entity_descriotion.key == "is_connected":
	self._attr_is_on = bool(status.is_linked)
```

Reply:
Or use subclasses with a different update method

Reply:
The idea was to allow space for future implementation, as per sensor.py.
Anyway, will update code.

Simone

Reply:
Yeah, I looked this is done in that as well (which also should be refactored, but not in this PR)
Subclasses are the common way to manage this, but can also be done with if's in the update method if you only want one class

Reply:
Code updated